nixos/modules/services/web-apps/pr-tracker.nix

```
@@ -0,0 +1,315 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) mkIf types;
+  inherit (lib.lists) optionals;
+  inherit (lib.attrsets) optionalAttrs;
+  inherit (lib.strings) concatStringsSep escapeShellArg;
+  inherit (lib.options) mkEnableOption mkOption mkPackageOption;
+
+  cfg = config.services.pr-tracker;
+in
+{
+  options.services.pr-tracker = {
+    enable = mkEnableOption "pr-tracker";
+
+    package = mkPackageOption pkgs "pr-tracker" { };
+
+    githubApiTokenFile = mkOption {
+      type = types.path;
+      example = "/run/secrets/gh-token";
+
+      description = ''
+        Path to a file containing your GitHub API token like so:
+
+        ```env
+        ghp_...
+        ```
+
+        ::: {.note}
+        The contents of this file will be the stdin of pr-tracker.
+        :::
+      '';
+    };
+
+    nixpkgsClone = {
+      cloneDir = mkOption {
+        type = types.path;
+        example = "/home/nixos/git/nixpkgs";
+
+        default = "${cfg.dataDir}/nixpkgs";
+        defaultText = "/var/lib/pr-tracker/nixpkgs";
+        description = ''
+          The path to the cloned nixpkgs pr-tracker will use.
+
+          ::: {.note}
+          If left as the default value this repo will automatically be cloned before
+          the pr-tracker server starts, otherwise you are responsible for ensuring
+          the directory exists with appropriate ownership and permissions.
+          :::
+        '';
+      };
+
+      remote = mkOption {
+        type = types.str;
+        example = "upstream";
+
+        default = "origin";
+        description = ''
+          The remote name in the repository corresponding to upstream Nixpkgs.
+        '';
+      };
+
+      managedByModule = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether you want this service to manage a clone of the nixpkgs
+          repo in `cfg.dataDir`.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        example = "15min";
+
+        default = "30min";
+        description = ''
+          How often to fetch nixpkgs if `cfg.nixpkgsClone.managedByModule` is true.
```

```suggestion
          How often to fetch nixpkgs if `services.pr-tracker.nixpkgsClone.managedByModule` is true.
```

we should also mention the format here. Probably systemd timer.